base_layer/core/src/chain_storage/test/chain_storage.rs

```diff
@@ -620,10 +620,10 @@ fn restore_mmr() {
     let rp_mmr_state = store.fetch_mmr_base_leaf_nodes(MmrTree::RangeProof, 0, 100).unwrap();
     let header_mmr_state = store.fetch_mmr_base_leaf_nodes(MmrTree::Header, 0, 100).unwrap();
 
-    assert_eq!(utxo_mmr_state.total_leaf_count, 9);
-    assert_eq!(kernel_mmr_state.total_leaf_count, 3);
-    assert_eq!(rp_mmr_state.total_leaf_count, 9);
-    assert_eq!(header_mmr_state.total_leaf_count, 2);
+    assert_eq!(utxo_mmr_state.total_leaf_count, 10);
+    assert_eq!(kernel_mmr_state.total_leaf_count, 4); // Todo This might be wrong, verify  1 + 3 + 2 +2  = 8
```

Comment:
```suggestion
    assert_eq!(kernel_mmr_state.total_leaf_count, 4); 
```
Only base mmr so 1(GB) + 3(B1)=4